packages/typespec-ts/src/utils/clientUtils.ts

```diff
@@ -0,0 +1,16 @@
+import {
+  SdkClient,
+  SdkContext,
+  listClients
+} from "@azure-tools/typespec-client-generator-core";
+
+export function getRLCClients(dpgContext: SdkContext): SdkClient[] {
+  const clientsInSdk = listClients(dpgContext);
+  const clients = listClients(dpgContext, false);
+  // to respect the boundaries that is defined by one @client
+  if (clientsInSdk.length === 1) {
+    return clientsInSdk;
+  }
+  // return the original @service clients
+  return clients;
```

Comment:
In light of the feedback on https://github.com/Azure/typespec-azure/pull/3190, I think we should avoid the change to listClients and do something like this:


```suggestion
  const clientsInSdk = listClients(dpgContext);
  // to respect the boundaries that is defined by one @client
  if (clientsInSdk.length === 1) {
    return clientsInSdk;
  }
  
  // Otherwise use service as the client
  const services = listServices(context.program);

  return services.map((service) => {
    const clientName = service.type.name + "Client";
    return {
      kind: "SdkClient",
      name: clientName,
      service: service.type,
      type: service.type,
    };
  });
```